packages/block-library/src/term-template/index.php

```diff
@@ -39,13 +40,15 @@ function render_block_core_term_template( $attributes, $content, $block ) {
 		'exclude'    => $query['exclude'] ?? array(),
 	);
 
-	// Handle parent.
-	if ( ! empty( $query['hierarchical'] ) && isset( $query['parent'] ) ) {
-		$query_args['parent'] = $query['parent'];
-	} elseif ( ! empty( $query['hierarchical'] ) ) {
+	if ( 'top-level' === $terms_to_show ) {
 		$query_args['parent'] = 0;
-	} elseif ( isset( $query['parent'] ) ) {
-		$query_args['parent'] = $query['parent'];
+	} elseif ( 'subterms' === $terms_to_show ) {
+		// Check if we're in a taxonomy archive context
+		if ( is_tax( $query_args['taxonomy'] ) ) {
+			// Get the current term ID from the queried object
+			$current_term_id      = get_queried_object_id();
+			$query_args['parent'] = $current_term_id;
```

Comment:
We could potentially make this more defensive, but I'm not sure it's necessary:

```suggestion
			if ( $current_term_id && $current_term_id > 0 ) {
				$query_args['parent'] = $current_term_id;
			}
```

Reply:
Yeah I think it makes sense!

Reply:
Done in https://github.com/WordPress/gutenberg/pull/71633/commits/b7b30d61c244a863bf979730d303b4f43b6a3517